torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -146,33 +146,26 @@ inline Tensor as_view(const Tensor & base, const Tensor & tensor, bool is_bw_dif
   // in the default_included_set, so that view ops on inference tensor doesn't
   // have to go through as_view even outside InferenceMode.
   if (base.unsafeGetTensorImpl()->is_inference_tensor()) return tensor;
-  if (!isForwardADEnabled()) {
-    // Fast codepath for backward only code
-    // It is useful as it avoids the creation of the temporary c10<optional> which makes
-    // a significant difference when measuring instruction count for a single "t.view(-1)" call from c++.
-    if (is_bw_differentiable) {
-      auto diff_view_meta = torch::autograd::impl::get_view_autograd_meta(base);
-      if (diff_view_meta && diff_view_meta->has_bw_view()) {
-        const auto& base_bw_info = diff_view_meta->get_backward_view();
-        creation_meta = propagate_creation_meta(diff_view_meta->get_creation_meta(), creation_meta);
-        return make_variable_differentiable_view(tensor, base_bw_info.chain(base, tensor, view_func),
-                                                 c10::nullopt, creation_meta, allow_tensor_metadata_change);
-      } else {
-        return make_variable_differentiable_view(tensor, ViewInfo(base, view_func),
-                                                 c10::nullopt, creation_meta, allow_tensor_metadata_change);
-      }
+
+  auto diff_view_meta = torch::autograd::impl::get_view_autograd_meta(base);
+
+  // Special case when view info can be shared for forward and backward differentiable views
```

Comment:
Is it a special case? Isn't this supposed to be the common case?

Reply:
It is the common case indeed.
I meant special case compared to the code that comes after that could handle all the cases.

Let me update the comment to make that distinction clearer.